velox/functions/sparksql/Rand.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/Macros.h"
+
+namespace facebook::velox::functions {
+
+template <typename T>
+struct RandFunction {
+  static constexpr bool is_deterministic = false;
+
+  std::optional<std::mt19937> randGenWithSeed;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& /*config*/,
+      const int64_t* seed) {
+    if (seed == nullptr) {
+      VELOX_USER_FAIL("Spark rand/random function requires a constant seed.");
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE void call(double& result) {
+    result = folly::Random::randDouble01();
+  }
+
+  FOLLY_ALWAYS_INLINE void callNullable(double& result, const int64_t* seed) {
+    if (!randGenWithSeed.has_value()) {
```

Comment:
Any particular reason not to move this logic to 'initialize'? This would remove an 'if' from per-row processing.

Reply:
Thanks for your comment!
Here, I want to distinguish null constant input (int64 type) and non-null constant input. Can we know whether the constant input is null in initialize method?